batch/batch/worker/worker.py

```diff
@@ -537,9 +536,7 @@ async def pull():
                     await self._ensure_image_is_pulled()
                 elif self.is_public_image:
                     await self._ensure_image_is_pulled(auth=self._batch_worker_registry_credentials)
-                elif self.image_ref_str == BATCH_WORKER_IMAGE and isinstance(
-                    self.credentials, (JVMUserCredentials, CopyStepCredentials)
-                ):
+                elif self.image_ref_str == BATCH_WORKER_IMAGE and self.hail_identity is None:
```

Comment:
should this say `is not None`?

Reply:
I'm not sure I follow what the previous if check was checking.

Reply:
I believe my change is consistent, but whether that clause needs to exist at all is a good question. The previous code regarding `JVMUserCredentials` and `CopyStepCredentials` allowed input/output containers and JVM containers, which we determine should use the extant batch worker image, to skip an auth check on the image. I originally changed that field from `Union[CloudUserCredentials, JVMUserCredentials, CopyStepCredentials]` to just `Optional[str]` so the `JVMUserCredentials` and `CopyStepCredentials` case was replaced with `None`.